docs/reference/using/js/client.rst

```diff
@@ -743,6 +743,33 @@ Client Reference
               // ...
             });
 
+    .. js:method:: withWarningHandler(handler: (warnings: errors.GelError[]) => void): Client
+
+        Returns a clone of the ``Client`` instance with the specified warning handler. Some queries may generate warnings while still returning a result. The ``handler`` function will be called with an array of ``GelError`` objects whenever the client encounters warnings during query execution.
```

Comment:
> Returns a clone

Do we have a dedicated section where we explain our "cloning"? IMO every time we mention cloning it should be linked to that page.

Also I'm not sure I like "cloning" 100% -- "deriving" might be a better word (but also it's a more complex word...) Thoughts?

Reply:
I went with the existing language from the rest of these methods, but it would be helpful to explain this more thoroughly and use consistent language, agreed.

Reply:
I'm going to keep what we have here and follow up if/when we decide to clarify (or change!) this behavior elsewhere in the docs.